rebac-admin-backend/v1/identities.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1
+
+import (
+	"encoding/json"
+	"net/http"
+
+	r "github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// GetIdentities Get list of identities.
+// (GET /identities)
+func (h handler) GetIdentities(w http.ResponseWriter, req *http.Request, params r.GetIdentitiesParams) {
+	identities, err := h.Identities.ListIdentities(&params)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := r.GetIdentitiesResponse{
+		Data:   identities.Data,
+		Status: 200,
+	}
+
+	writeResponse(w, 200, response)
+}
+
+// PostIdentities Add an identity.
+// (POST /identities)
+func (h handler) PostIdentities(w http.ResponseWriter, req *http.Request) {
+	identity := new(r.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
+	if err != nil {
+		writeErrorResponse(w, err)
+		return
+	}
+
+	identity, err = h.Identities.CreateIdentity(identity)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 201, identity)
+}
+
+// DeleteIdentitiesItem Remove an identity.
+// (DELETE /identities/{id})
+func (h handler) DeleteIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	_, err := h.Identities.DeleteIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	w.WriteHeader(200)
+}
+
+// GetIdentitiesItem Get a single identity.
+// (GET /identities/{id})
+func (h handler) GetIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity, err := h.Identities.GetIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 200, identity)
+}
+
+// PutIdentitiesItem Update an identity.
+// (PUT /identities/{id})
+func (h handler) PutIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity := new(r.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
+	if err != nil {
+		writeErrorResponse(w, err)
+		return
+	}
+
+	if id != *identity.Id {
+		writeErrorResponse(w, NewValidationError("Identity ID from path does not match the Identity object"))
+		return
+	}
+
+	identity, err = h.Identities.UpdateIdentity(identity)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 200, identity)
+}
+
+// GetIdentitiesItemEntitlements List entitlements the identity has.
```

Comment:
// GetEntitlements returns the list of entitlements for an identity identified by the provided ID.